eos-tech-support/eos-write-live-image

```diff
@@ -186,106 +237,139 @@ check_exists "$BOOT_ZIP.asc" "bootloader bundle signature"
-        exit 1
-    fi
-    echo $MOUNT_POINT
-}
+echo
+echo "Copying files"
 
-udisks_unmount() {
-    udisksctl unmount -b "$1" --no-user-interaction >&2
-}
+DIR_IMAGES_ENDLESS="${DIR_IMAGES}/endless"
+mkdir "$DIR_IMAGES_ENDLESS"
+unzip -d "${DIR_IMAGES_ENDLESS}" "${BOOT_ZIP}" "grub/*"
 
-if [ ! "$MBR" ]; then
-    MOUNT_EFI=$(udisks_mount "$DEVICE_EFI")
-    unzip -d "${MOUNT_EFI}" "${BOOT_ZIP}" "EFI/*"
-    udisks_unmount "$DEVICE_EFI"
+if [ ! -z "$DIR_EFI" ]; then
+    unzip -d "${DIR_EFI}" "${BOOT_ZIP}" "EFI/*"
 fi
 
-MOUNT_IMAGES=$(udisks_mount "$DEVICE_IMAGES")
-MOUNT_IMAGES_ENDLESS="${MOUNT_IMAGES}/endless"
-mkdir "$MOUNT_IMAGES_ENDLESS"
-unzip -d "${MOUNT_IMAGES_ENDLESS}" "${BOOT_ZIP}" "grub/*"
-
 if [ ! "$WRITABLE" ]; then
     cp "${BOOT_ZIP}" "${BOOT_ZIP}.asc" "${EXTRACTED_SIGNATURE}" \
-        "$MOUNT_IMAGES_ENDLESS/"
-    echo "$(basename "${OS_IMAGE%.?z}")" > "${MOUNT_IMAGES_ENDLESS}/live"
-    cp "$WINDOWS_TOOL" "$MOUNT_IMAGES/"
+        "$DIR_IMAGES_ENDLESS/"
+    echo "$(basename "${OS_IMAGE%.?z}")" > "${DIR_IMAGES_ENDLESS}/live"
+    cp "$WINDOWS_TOOL" "$DIR_IMAGES/"
 fi
 
-"$EOS_WRITE_IMAGE" --removable "$OS_IMAGE" "-" > "${MOUNT_IMAGES_ENDLESS}/endless.img"
-udisks_unmount "$DEVICE_IMAGES"
+"$EOS_WRITE_IMAGE" --removable "$OS_IMAGE" "-" > "${DIR_IMAGES_ENDLESS}/endless.img"
 
-if [ "$MBR" ]; then
-    # Bootstrap code, which will jump to sector 1
-    unzip -p "${BOOT_ZIP}" "ntfs/boot.img" | dd of="${DEVICE}" bs=446 count=1
-    udevadm settle  # udev recreates device files after any write to the device
-    # The rest of GRUB goes after the MBR and before the first partition.
-    unzip -p "${BOOT_ZIP}" "ntfs/core.img" | dd of="${DEVICE}" bs=512 seek=1
-else
-    # Bootstrap code, built to jump to the offset of BIOS boot partition
-    unzip -p "${BOOT_ZIP}" "live/boot.img" | dd of="${DEVICE}" bs=446 count=1
-    udevadm settle  # udev recreates device files after any write to the device
-    # The rest of GRUB goes into a dedicated BIOS-boot partition
-    unzip -p "${BOOT_ZIP}" "live/core.img" | dd of="${DEVICE_BIOS}" bs=512
+echo
+echo "Finalizing image"
+
+if [ "$ISO" ]; then
+    DIR_EFI_SIZE=$(du -s ${DIR_EFI} | cut -f1)
+    ESP_SIZE=$(( (${DIR_EFI_SIZE} + 1024) / 1024 * 1024 ))
+
+    # Create ESP and copy the EFI content
+    truncate -s ${ESP_SIZE}K ${DIR_IMAGES_ENDLESS}/efi.img
+    mkdosfs ${DIR_IMAGES_ENDLESS}/efi.img
+    mcopy -s -i ${DIR_IMAGES_ENDLESS}/efi.img ${DIR_EFI}/EFI '::/'
+
+    # Generate the ISO image
+    unzip -d "${DIR_IMAGES_ENDLESS}/grub/i386-pc/" "${BOOT_ZIP}" "iso/*"
+    xorriso -as mkisofs \
+            -graft-points \
+            -b endless/grub/i386-pc/iso/eltorito.img \
+            -no-emul-boot \
+            -boot-load-size 4 \
+            -boot-info-table \
+            --grub2-boot-info \
+            --grub2-mbr ${DIR_IMAGES_ENDLESS}/grub/i386-pc/iso/boot_hybrid.img \
+            --efi-boot endless/efi.img \
+            -efi-boot-part \
```

Comment:
What's going on with the weird combination of `-` and `--` prefixes on options? An idiosyncrasy of `xorriso`?